test/test_router.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	c "github.com/flynn/go-check"
+)
+
+type RouterSuite struct {
+	Helper
+}
+
+var _ = c.ConcurrentSuite(&RouterSuite{})
+
+func (s *RouterSuite) TestAdditionalHttpPorts(t *c.C) {
+	// boot 1 node cluster
+	x := s.bootCluster(t, 1)
+	defer x.Destroy()
+
+	// Test that setting added HTTP and HTTPS ports succeeds
+	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTP_PORTS=8080"), Succeeds)
+	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTPS_PORTS=8081"), Succeeds)
+
+	// check a non-routed HTTP request to an additional port fails
+	req, err := http.NewRequest("GET", "http://dashboard."+x.Domain+":8080", nil)
+	t.Assert(err, c.IsNil)
+	req.SetBasicAuth("", x.Key)
+	res, err := http.DefaultClient.Do(req)
+	t.Assert(err, c.IsNil)
+	t.Assert(res.StatusCode, c.Equals, http.StatusNotFound)
+
+	// add a controller route on the new port
+	t.Assert(x.flynn("/", "-a", "dashboard", "route", "add", "http", "-s", "dashboard-web", "-p", "8080", "dashboard."+x.Domain), Succeeds)
+
+	// FIXME: This delay should be unnecessary, but appears to be vital. The following test
+	// has been repeated, to ensure that it is not the problem; when the delay is removed,
+	// the repetition may be as well.
+	time.Sleep(1 * time.Second)
+	for i := 0; i < 20; i++ {
```

Comment:
That helper currently checks whether the route exists in the flynn route output; it does nothing to test whether the port is available. I could abstract it out, but I feel like the overlap is actually fairly minimal.